hmy/api_backend.go

```diff
@@ -0,0 +1,187 @@
+package hmy
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/bloombits"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/harmony-one/harmony/accounts"
+	"github.com/harmony-one/harmony/api/proto"
+	"github.com/harmony-one/harmony/core"
+	"github.com/harmony-one/harmony/core/state"
+	"github.com/harmony-one/harmony/core/types"
+)
+
+// APIBackend An implementation of Backend. Full client.
+type APIBackend struct {
+	hmy *Harmony
+}
+
+// ChainDb ...
```

Comment:
ek in another PR complains about this type of comments. can you write all full comments? i totally agree in our code base there are other similar comments.

Reply:
Will fix later